### PR TITLE
feat: Make all Worker channels metered

### PR DESCRIPTION
**Pull request by huitseeker**
This is an extension of #682 to the worker

### CODE REVIEW

**Review by gdanezis — APPROVED** [review]
Awesome conversion!

File: worker/src/synchronizer.rs [hunk 1]
```diff
@@ -164,7 +162,7 @@ impl Synchronizer {
 
                             // Add the digest to the waiter.
                             let deliver = digest;
-                            let (tx_cancel, rx_cancel) = channel(1);
+                            let (tx_cancel, rx_cancel) = mpsc::channel(1);
```
Comment (akichidis) [inline worker/src/synchronizer.rs:165]: Ah probably those could be oneshot channels? (not concern of this PR of course - but something to keep in mind)

**Review by akichidis — APPROVED** [review]
Great work @huitseeker , thanks for doing this!

File: worker/src/synchronizer.rs (hunk 1 above)
Comment (huitseeker) [reply worker/src/synchronizer.rs:165]: They usually can't: this is a cancellation handler, and the branches of the `tokio::select` that GC and those that unpack items in waiting race each other. The GC might then try to cancel an item that's already delivered (and the delivery sinks the receiving end of the channel). With a oneshot channel, that may lead to a panic.

**Pull request merged**